content/docs/overview.md

```diff
@@ -1792,6 +1792,61 @@ f: Foo(T, N);
 #assert(size_of(f) == (N+N-2)*size_of(T));
 ```
 
+## Custom iterator functions
+
+Custom iterator functions can be used to iterate over anything you want.
+
+A custom iterator function should take a pointer to a struct keeps track of the state of the iteration.
+
+The custom iterator function should return 3 values. The next value to be given to the loop, the next index for the loop, and a boolean to signify if the loop should keep running.
+
+```odin
+//Iterate a linked list
+Node :: struct {
+    value: int,
+    next: ^Node
+}
+
+MyIterator :: struct {
+  index: int,
+  current: Node
+}
+//The custom iterator function can be named anything.
+custom_iterator_function :: proc(it: ^MyIterator) -> (value: Node, index: int, ok: bool) {
+    value = it.current;
+    index = it.index;
+    if value.next != nil {
+      it.current = it.current.next;
+      ok = true;
+      it.index += 1;
+    }
+    return;
+}
+
+main :: proc () {
+    head : Node = {value=10, next=nil};
+    tail : Node = {value=10, next=nil};
+    
+    head.next = tail;
+    
+    it : MyIterator = {index = 0; current=head};
+    
+    for node in custom_iterator_function(&it) {
+      fmt.println(node);
+    }
+}
+```
```

Comment:
I am not sure a linked list is the best example especially since honestly prefer linked lists to be C-style loops e.g.
```odin
for node := head; node != nil; node = node.next {
    ...
}
```

However, I am not sure what the best example would be.

Reply:
An iterator that visits the elements in a random order? Albeit not perhaps the _most_ realistic example...